### PR TITLE
Misc @bolt/core + @bolt/config Upgrades and Maintenance 

### PR DESCRIPTION
**Pull request by sghoweri**
- Upgrade Babel config to use latest versions of Babel + Babel Plugins
- Remove Babel plugins no longer required
- Update Webpack Babel Loader to use latest version that supports ^
- Fix error in eslint config, clean up a few files based on config settings
- Upgrade polyfill loader to use latest technique that ships web component polyfills based on exactly combination of shims the browser needs (vs just a once-size-fits-all bundle)
- Fix JS error ('Reflect' is undefined) by reverting the base Class prototype setup:
```
export function withPreact(Base = class extends HTMLElement { }){
```
is now (back to)
```
export function withPreact(Base = HTMLElement) {
```

### CODE REVIEW

**Comment by sghoweri** [comment]
PS. Haven't confirmed this yet but I THINK splitting out https://github.com/bolt-design-system/bolt/pull/498 into it's own separate PR is the root cause for the Travis error. Guessing that the build error should go away once that other PR gets merged in (or I pull that work into this PR)

File: apps/pattern-lab/src/styles/_docs.scss [hunk 1]
```diff
@@ -60,14 +60,42 @@
...
 
 .c-bolt-docs__page-content {
-  grid-column: 1/span 3;
+  -ms-grid-row: 3;
+  -ms-grid-row-span: 1;
   grid-row: 3/span 1;
+
+  -ms-grid-column: 1;
+  -ms-grid-column-span: 3;
+  grid-column: 1/span 3;
```
Comment (EvanLovely) [inline apps/pattern-lab/src/styles/_docs.scss]: - [ ] Remove vender prefixes; we have autoprefixer.

File: packages/core/polyfills/polyfill-loader.js [hunk 2]
```diff
@@ -23,26 +27,47 @@ if (window.NodeList && !NodeList.prototype.forEach) {
...
+  polyfills = ['lite'];
+}
+
 
+const webComponentPolyfillPath = `bolt-webcomponents-${polyfills.join('-')}.js`;
 
 
-export const polyfillLoader = new Promise(function (resolve, reject) {
-  if (window.customElements !== undefined) {
+export const polyfillLoader = new Promise((resolve) => {
```
Comment (EvanLovely) [inline packages/core/polyfills/polyfill-loader.js]: This promise should also have a `reject`, even if that just calls `console.error('something did not load...')`

- [ ] Add `reject` to pollyfill loader

File: apps/pattern-lab/src/styles/_docs.scss (hunk 1 above)
Comment (sghoweri) [reply apps/pattern-lab/src/styles/_docs.scss]: 100% agree, except this one specific use case (yes, yes I know, I should have added a comment on this).

The old draft version of CSS Grid that IE 10 and IE 11 shipped doesn't get transpiled as you'd expect via Autoprefixer, hence why the docs layout in PL looks the way that it does currently in IE11... 🤦‍♂️

image

https://rachelandrew.co.uk/archives/2016/11/26/should-i-try-to-use-the-ie-implementation-of-css-grid-layout/

File: packages/core/polyfills/polyfill-loader.js (hunk 2 above)
Comment (sghoweri) [reply packages/core/polyfills/polyfill-loader.js]: @EvanLovely doesn't [this](https://github.com/bolt-design-system/bolt/blob/6536e39dc036ec3675289d83f164c62610cffd91/packages/core/polyfills/polyfill-loader.js#L83) accomplish that? Or am I getting my chained promises mixed up here?

**Pull request merged**